File: programs/saveTable.R
# saveTable("test.txt", "<tab:testR1>", matrix(1:4, 2, 2))
# saveTable("test.txt", "<tab:testR2>", matrix(runif(4), 2, 2))

saveTable <- function (outfile, tag, outmatrix) {
    cat(tag,
        sep    = "\n",
        file   = outfile,
        append = TRUE)

    write.table(outmatrix,
                file      = outfile,
                sep       = "\t",
                append    = TRUE,
                quote     = FALSE,
                col.names = FALSE,
                row.names = FALSE)
}

File: programs/saveTable.py
# saveTable("test.txt", "<tab:testPython1>", np.arange(4).reshape(2, 2))
# saveTable("test.txt", "<tab:testPython2>", np.random.uniform(size = (2, 2)))

from os import linesep
import numpy as np


def saveTable(outfile, tag, outmatrix):
    with open(outfile, "a") as fh:
        fh.write(tag)
        fh.write(linesep)
        np.savetxt(fh, outmatrix, delimiter = '\t')
